src/common/exception/GraknClientException.jl

```diff
@@ -2,13 +2,35 @@
 
 mutable struct GraknClientException <: Exception
     error_message::T where {T<:Union{<:AbstractGeneralError,<:Nothing}}
-    params
+    params::Optional{Tuple}
     individual_message::Union{Nothing, String}
-    cause::R where {R<:Exception}
+    cause::R where {R<:Union{Nothing, Exception}}
+end
+
+Base.show(io::IO, grakn_excption::GraknClientException) = Base.print(io,grakn_excption)
+function Base.print(io::IO, grakn_excption::GraknClientException)
+    err_message = string(grakn_excption.error_message)
+    joined_params = ""
+    if grakn_excption.params !== nothing
+        if !isempty(grakn_excption.params)
+            replace_item = "_error_item"
+            err_message = replace(err_message, replace_item=>string(grakn_excption.params[1]))
+            joined_params = join(grakn_excption.params, "\n")
+        end
+    end
+    str = "GraknClientException:
+        message: $(err_message)
+        params: $(joined_params)
+        remark: $(grakn_excption.individual_message)
+        cause: $(Base.show(grakn_excption.cause))"
+
+    print(io,str)
+    return nothing
 end
 
 function GraknClientException(err::Type{T}, parameters...) where {T<:AbstractGeneralError}
```

Comment:
Again, unless you are going to use the type variable within the function, don't bother with a `where` clause.